005_team_3_agent_design/005_team_3_agent_design.tex

```diff
@@ -1 +1,129 @@
+    \item \texttt{hpAbove int}: stores the HP our neighbour above last told us they had
+    \item \texttt{hpBelow int}: stores the HP our neighbour below last told us they had
+  \end{enumerate}
+
+\subsection{Agent decisions}
+Agent 3 will be able to take decisions when receiving messages or after signing treaties and store these until they eat. This information is used to symbolize a predetermined decision has been taken instead of “impulsively” eating food depending solely on our hunger and state of mind.
+\begin{enumerate}
+    \item \texttt{foodToEat int}: stores how much food we have decided to eat in the next eating period.
+    \item \texttt{foodToLeave int}: how much food we have decided to leave. It is necessary in case we want to: E.g eat 6 food and leave at least 10 food but when the platform arrives we see that it has 13 food. Both statements cannot be fulfilled and so we must prioritise one.
+\end{enumerate}
+
+\subsection{Agent variables}
+Agent 3 will act differently depending on three different variables that define them, which will give the agent different personalities.
+\begin{enumerate}
+    \item \texttt{stubbornness int}: defines the likelihood of the agent to read a message. E.g: Stubbornness = 20 means there is a 20\% chance that the agent will ignore the message.
+    \item \texttt{morality int}: defines the willingness of the agent to help others, \cite{waller_2015}, in other words, how much you care about other agents in the tower. 
+    \item \texttt{mood int}: defines how likely the agent is to do things, it will affect its decision-making process.
+\end{enumerate}
+For the purpose of this agent, we will define personality as the individual differences in characteristic patterns of thinking, feeling, and behaving, as defined by the American Psychology Association. The traits that define an agent's personality are mutable, making our agent adapt its personality depending on the circumstances. 
+
+\subsection{Agent generation}
+%%TODO: have to add the code snippet
+The function generates a new agent by initializing its variables, knowledge and decisions. All knowledge and decisions are defined initially as empty except for our last HP, which is initialized as 100 (as all agents are). \par
+The variables of our agent, which define the personality it has, are randomly allocated to generate different agents. All three variables are defined to be in the range 0 to 100. Therefore morality and mood are randomly allocated to a value between 0 and 100. To ensure our agent is not completely deaf from the start we limit the initial stubbornness to 75. This does not limit the agent once it starts interacting and taking decisions. Which means it could possibly end up with a stubbornness value of 100 if the situations it goes through are auspicious for it.
+
+\subsection{Agent Strategy}
+Our agent follows the following strategy depending on the three agent variables:
```

Comment:
I wouldn't use "our" in an academic report. Use "this agent" or "it" contextually. You have a lot of uses of "our agent" and sometimes "the agent," (which would be correct). Best not to mix them up

Reply:
Done